backend/src/index.js

```diff
@@ -8,11 +8,17 @@
 const PORT = process.env.PORT || 3000;
 
 const app = require( 'express' )();
+const { promisify } = require( 'util' );
 const { Server } = require( 'http' );
 const http = new Server( app );
 const io = require( 'socket.io' )( http );
+const Redis = require( 'ioredis' );
+const redisAdapter = require( 'socket.io-redis' );
+io.adapter( redisAdapter( { host: 'localhost', port: 6379 } ) );
 
-const db = new Map();
+const client = new Redis();
+const fetchUsers = promisify( client.get ).bind( client );
```

Comment:
https://github.com/luin/ioredis - as I see Redis supports Promises out of the box.

Reply:
Good point.